base/path.jl

```diff
@@ -248,16 +248,17 @@ function splitpath(p::String)
     return out
 end
 
-joinpath(path::AbstractString)::String = path
-
 if Sys.iswindows()
 
-function joinpath(path::AbstractString, paths::AbstractString...)::String
-    result_drive, result_path = splitdrive(path)
+function joinpath(paths::Union{Tuple, AbstractVector{<:AbstractString}})::String
+    isempty(paths) && return path_separator
+    typeassert(paths[1], AbstractString)
```

Comment:
You'll want to do something like `paths[1] isa AbstractString || throw(ArgumentError("path component is not a string: $(repr(paths[1]))`. Since you'll be repeating that, you may want a helpful function for it.